src/selectedtests/injection_helper.py

```diff
@@ -0,0 +1,22 @@
+import inspect
+from typing import Any
+
+import inject
+
+
+def remap_for_injection(func):
+    specs = inspect.getfullargspec(func)
+    arg_map = {arg_name: specs.annotations.get(arg_name, Any) for arg_name in specs.args}
+
+    def partial(*args, **kwargs):
+        args = list(reversed(args))
+        new_args = []
+        injector = inject.get_injector_or_die()
+        for arg_name, anno in arg_map.items():
+            if anno in injector._bindings:
+                new_args.append(injector.get_instance(anno))
+            else:
+                new_args.append(args.pop())
+        return func(*new_args, **kwargs)
+
+    return partial
```

Comment:
This is to avoid refactoring signatures like `func(dep, arg)` to `func(arg, dep)`. This will inject `dep` even if it comes before `arg`, if something is registered for that annotation.